### PR TITLE
[CDAP-16850] Changes Avro schema types from string to enum

### PR DESCRIPTION
**Pull request by ajainarayanan**

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts [hunk 1]
```diff
@@ -184,24 +187,29 @@ const complexUnionField: IUnionField = {
...
+    AvroSchemaTypesEnum.LONG,
+    AvroSchemaTypesEnum.STRING,
+    AvroSchemaTypesEnum.TIME,
   ].indexOf(type) !== -1;
 if (Array.isArray(complexUnionField.type)) {
   const map1 = complexUnionField.type
     .filter((t) => typeof t !== 'string')
-    .find((t: IComplexType) => !Array.isArray(t.type) && t.type === 'map') as IMapFieldBase;
+    .find(
+      (t: IComplexType) => !Array.isArray(t.type) && t.type === AvroSchemaTypesEnum.MAP
```
Comment (jennac3) [inline cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts:205]: We can also do `t.type !== AvroSchemaTypesEnum.ARRAY && t.type === AvroSchemaTypesEnum.MAP`

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts [hunk 2]
```diff
@@ -184,24 +187,29 @@ const complexUnionField: IUnionField = {
...
+    ) as IMapFieldBase;
   let fieldsInRecords;
-  if (!Array.isArray(map1.keys) && typeof map1.keys === 'object' && map1.keys.type === 'record') {
+  if (
+    !Array.isArray(map1.keys) &&
```
Comment (jennac3) [inline cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts:209]: Same comment as above!

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts [hunk 3]
```diff
@@ -19,30 +19,32 @@ import {
...
+  AvroSchemaTypesEnum.DECIMAL,
+  AvroSchemaTypesEnum.DOUBLE,
+  AvroSchemaTypesEnum.FLOAT,
+  AvroSchemaTypesEnum.INT,
+  AvroSchemaTypesEnum.LONG,
+  AvroSchemaTypesEnum.STRING,
+  AvroSchemaTypesEnum.TIME,
+  AvroSchemaTypesEnum.TIMESTAMPMICROS,
+  AvroSchemaTypesEnum.DATE,
+  'AvroSchemaTypesEnum.TIMEMICROS',
```
Comment (jennac3) [inline cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts]: Any reason why this is wrapped in string?

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts:205]: not quite. Array.isArray is checking whether the argument is an array, not the string 'array'

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts (hunk 3 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts]: Thats a mistake. Fixed it. Thanks for catching this.

**Review by elfenheart — APPROVED** [review]
👍  LGTM

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts [hunk 4]
```diff
@@ -19,30 +19,32 @@ import {
...
-  'union',
-  'boolean',
-  'bytes',
-  'date',
-  'decimal',
-  'double',
-  'float',
-  'int',
-  'long',
-  'number',
```
Comment (njbriggs) [inline cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts:39]: Did you mean to drop `number`?

**Review by njbriggs — COMMENTED** [review]
1 question, everything else looks good.

File: cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts (hunk 4 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SchemaHelpers.ts:39]: Yeah. I don't know why I added it. `number` is not an avro type.

**Comment by ajainarayanan** [comment]
Closing this PR and cherry-picking 0843b32 commit to base branch

**Pull request closed without merging**